service/src/main/java/bio/terra/cbas/dao/RunDao.java

```diff
@@ -20,4 +30,37 @@ public int createRun(Run run) {
             + " values (:id, :engineId, :runSetId, :entityId, :submissionTimestamp, :status)",
         new BeanPropertySqlParameterSource(run));
   }
+
+  public List<Run> retrieve() {
+    String sql =
+        "SELECT * FROM run INNER JOIN run_set ON run.run_set_id = run_set.id"
+            + " INNER JOIN method ON run_set.method_id = method.id";
+
+    try {
+      return jdbcTemplate.query(sql, new RunMapper());
+    } catch (EmptyResultDataAccessException er) {
+      throw new RunQueryNotFoundException("Runs not found.");
```

Comment:
This doesn't feel quite right... if there are no runs then don't we get an empty result back? So I wonder what would cause this exception to happen?

Reply:
Right, I think I was thinking in the mindset of entering an id, and having that id match or not match data in the db, but we're returning everything, so this isn't really necessary. I'll remove on next commit.